test/test_optim.py

```diff
@@ -1704,38 +1706,44 @@ def _grad_scaling_autocast_fused_optimizers(self, optimizer_ctor, optimizer_kwar
             kwargs['lr'] = 1.0
         opt_control = optimizer_ctor(mod_control.parameters(), **kwargs)
 
-        scaler = torch.cpu.amp.GradScaler(init_scale=128.0)
+        scaler_scaling = torch.amp.GradScaler(device, init_scale=128.0)
+        scaler_control = torch.amp.GradScaler(device, init_scale=128.0)
+        tracker = TensorTracker()
         for input, target in data:
             opt_control.zero_grad()
-            with torch.autocast('cpu', dtype=torch.half):
+            with torch.autocast(device_type=device, dtype=torch.half):
                 output_control = mod_control(input)
                 loss_control = loss_fn(output_control, target)
-            scaler.scale(loss_control).backward()
-            scaler.step(opt_control)
-            scaler.update()
+            scaler_control.scale(loss_control).backward()
+            scaler_control.step(opt_control)
+            scaler_control.update()
 
             opt_scaling.zero_grad()
-            with torch.autocast('cpu', dtype=torch.half):
+            with torch.autocast(device_type=device, dtype=torch.half):
                 output_scaling = mod_scaling(input)
                 loss_scaling = loss_fn(output_scaling, target)
-            scaler.scale(loss_scaling).backward()
+            scaler_scaling.scale(loss_scaling).backward()
             if separate_unscale:
-                scaler.unscale_(opt_scaling)
-            scaler.step(opt_scaling)
-            scaler.update()
+                scaler_scaling.unscale_(opt_scaling)
```

Comment:
yikes, good catch! does this mean the original test in test_cuda is wrong today then? 

could we unify the test now?

Reply:
Yes, we should eventually move the test back to test_cuda.py under TestCudaOptim to keep gradscaling together but let's keep in test_optim.py for now as there's a lot of movement in that file recently :D